File: test/dump_config.pl
#!/usr/bin/perl

use strict;
use Config::IniFiles;

my $config = '/etc/blabgen/config.ini';
$config = '../conf/config.ini' if -r '../conf/config.ini'; # for debug
$config = 'conf/config.ini' if -r 'conf/config.ini'; # for debug

my $local_conf = '/etc/blabgen/local.ini';
$local_conf = '../conf/local.ini'
	if -r '../conf/local.ini'; # for debug
$local_conf = 'conf/local.ini' if -r 'conf/local.ini'; # for debug

my $config_obj = new Config::IniFiles(-file => $config);
die "no config" unless $config_obj;
if (-r $local_conf) {
	$config_obj = new Config::IniFiles(-file => $local_conf,
		-import => $config_obj);
}

for my $s (sort $config_obj->Sections) {
	for my $p (sort $config_obj->Parameters($s)) {
		print "$s\t$p\t", $config_obj->val($s, $p), "\n";
	}
}

File: test/dump_config.php
#!/usr/bin/php
<?php

$opts_tmp = array();
$config = '/etc/blabgen/config.ini';
$dev_config = __DIR__.'/../conf/config.ini';
if (is_readable($dev_config)) {
	$config = $dev_config;
}
print "Reading $config\n";
$opts = parse_ini_file($config, 1);

# global config that is local
$config = '/etc/blabgen/local.ini';
$dev_config = __DIR__.'/../conf/local.ini';
if (is_readable($dev_config)) {
	$config = $dev_config;
}
print "Reading local $config\n";
if (is_readable($config)) {
	$opts_tmp = parse_ini_file($config, 1);
	$opts = array_replace_recursive($opts, $opts_tmp );
}

// some configuration depends on remote hostname
$hn = $argv[1];
$config = '/etc/blabgen/host-'.$hn.'.ini';
$dev_config = __DIR__.'/../conf/host-'.$hn.'.ini';
if (is_readable($dev_config)) {
	$config = $dev_config;
}
if (is_readable($config)) {
	print "Reading host $config\n";
	$opts_tmp = parse_ini_file($config, 1);
	$opts = array_replace_recursive($opts, $opts_tmp);
}

foreach($opts as $k => $p) {
	foreach($opts[$k] as $kp => $v) {
		print "$k\t$kp\t$v\n";
	}
}
	
